src/App.tsx

```diff
@@ -58,36 +60,33 @@ export default class App extends Component<{}, { chosenNode: TreeNode<string>, v
     }
 
 
-
-    addCustom(event: any, node_key: string, value: string)
+    addCustom(event: any, node_id: string, value: string)
     {
-        this.state.ourTree.filter(n => n.data === node_key)[0].add(new TreeNode(value))
-
+        this.state.ourTree.filter(n => n.id === node_id)[0].add(new TreeNode(value))
         this.setState({
             ourTree: this.state.ourTree
         })
-
     }
-
     remove(event: any)
     {
         getParent(this.state.chosenNode, this.state.ourTree).removeNode(this.state.chosenNode)
 
         this.setState({
-            ourTree: this.state.ourTree
+            ourTree: this.state.ourTree,
+            chosenNode: this.state.ourTree
         })
     }
 
-    displayNode(event: any, node_key: string) 
+    selectNode(event: any, node_id: string) 
     {
         this.setState({
-            chosenNode: this.state.ourTree.filter(n => n.data === node_key)[0]
+            chosenNode: node_id === this.state.ourTree.id ? this.state.ourTree : this.state.ourTree.filter(n => n.id === node_id)[0]
```

Comment:
What does the check here is for?

Reply:
bump

Reply:
Earlier version did not catch the edge case of root node.